pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -463,7 +463,12 @@ public void unloadNamespaceBundlesGracefully() {
     }
 
     public CompletableFuture<Topic> getOrCreateTopic(final String topic) {
-        return getTopic(topic, true /* createIfMissing */).thenApply(Optional::get);
+    	return getOrCreateTopic(topic, true /* use allowAutoTopicCreation */);
+    }
+    
+    public CompletableFuture<Topic> getOrCreateTopic(String topic, boolean useAllowAutoTopicCreationSetting) {
```

Comment:
Instead of creating a new override for getOrCreateTopic, make getTopic public. 
Then the one place that calls ```getOrCreateTopic(..., false)``` can instead call ```getTopic(..., true)```, and getOrCreateTopic() can become
```
public CompletableFuture<Topic> getOrCreateTopic(final String topic) { 
             return getTopic(topic, pulsar.getConfiguration().isAllowAutoTopicCreation()).thenApply(Optional::get); 
}
```